internal/dataplane/parser/translate_ingress.go

```diff
@@ -58,8 +58,16 @@ func (p *Parser) ingressRulesFromIngressV1() ingressRules {
 
 	// Translate Ingress objects into Kong Services.
 	for _, service := range p.ingressesV1ToKongServices(ingressList, icp) {
-		result.ServiceNameToServices[*service.Name] = service
-		result.ServiceNameToParent[*service.Name] = service.Parent
+		serviceCopy := service
```

Comment:
Here you perform only shallow copy of https://github.com/Kong/kubernetes-ingress-controller/blob/0593166677bcb8a3a07c8b9cd5dd95d25d62169e/internal/dataplane/kongstate/service.go#L45-L61

Does it safe? Does it intentional?

Reply:
Here is just for avoiding lint problems.

Reply:
Oh, I see. You can rewrite it something like this, I think it is more idomatic
```go
	// Translate Ingress objects into Kong Services.
	servicesCache := p.ingressesV1ToKongServices(ingressList, icp)
	for i := range servicesCache {
		service := servicesCache[i]
		if p.featureFlags.RewriteURIs {
			if err := translators.MaybeRewriteURI(&service); err != nil {
				p.registerTranslationFailure(err.Error(), service.Parent)
				continue
			}
		}
		result.ServiceNameToServices[*service.Name] = service
		result.ServiceNameToParent[*service.Name] = service.Parent
	}
```